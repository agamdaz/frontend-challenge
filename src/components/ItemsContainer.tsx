import React from "react";
import {style} from "typestyle";
import {ListItem} from "./Item";

const itemsContainer = style({
  width: '50%',
  minHeight: '300px',
  margin: 0,
  padding: 0,
  border: '2px solid #000',
})

type ListContainerProps = {
  items: string[],
}

export const ItemsContainer = ({items}: ListContainerProps) => (
  <ul data-testid="items-container" className={itemsContainer}>
    {items.map((label, key) => (<ListItem key={key} label={label}/>))}
  </ul>
);
