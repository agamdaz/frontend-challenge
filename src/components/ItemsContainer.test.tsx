import React from 'react';
import { render, screen } from '@testing-library/react';
import { ItemsContainer } from "./ItemsContainer";

test('Renders ListContainer component with items', () => {
  render(<ItemsContainer items={['AnyFirstItem', 'AnySecondItem']}/>);
  expect(screen.getByTestId('items-container')).toMatchSnapshot();
});

test('Renders ListContainer component with no items', () => {
  render(<ItemsContainer items={[]}/>);
  expect(screen.getByTestId('items-container')).toMatchSnapshot();
});
