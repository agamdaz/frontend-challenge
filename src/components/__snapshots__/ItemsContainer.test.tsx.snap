// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders ListContainer component with items 1`] = `
<ul
  class="f1fmxlhp"
  data-testid="items-container"
>
  <li
    class="f1v5ywig"
    data-testid="list-item"
  >
    <button
      class="f1f3cgq1"
    >
      AnyFirstItem
    </button>
  </li>
  <li
    class="f1v5ywig"
    data-testid="list-item"
  >
    <button
      class="f1f3cgq1"
    >
      AnySecondItem
    </button>
  </li>
</ul>
`;

exports[`Renders ListContainer component with no items 1`] = `
<ul
  class="f1fmxlhp"
  data-testid="items-container"
/>
`;
